services/addons/tests/iss_health_check/test_iss_token.py

```diff
@@ -1,230 +1,482 @@
-    mock_check_if_secret_exists.return_value = True
-    mock_get_latest_secret_version.return_value = {
-        "name": "test-api-key-name_01234",
-        "token": "old-token",
-    }
-    mock_uuid.uuid4.return_value = 12345
-    mock_requests.post.return_value = mock_response
-    mock_response.json.return_value = {"Item": "new-iss-token"}
-
-    # Call function
-    expected_value = "new-iss-token"
-    actual_value = iss_token.get_token()
-
-    # Check if iss_token function calls were made correctly
-    mock_check_if_secret_exists.assert_called_with(
-        mock_sm_client, "iss-token-secret", "projects/test-proj"
-    )
-    mock_get_latest_secret_version.assert_called_with(
-        mock_sm_client, "iss-token-secret", "projects/test-proj"
-    )
-    mock_add_secret_version.assert_called_with(
-        mock_sm_client,
-        "iss-token-secret",
-        "projects/test-proj",
-        {"name": "test-api-key-name_12345", "token": expected_value},
-    )
-
-    # Check if HTTP requests were made correctly
-    expected_headers = {"x-api-key": "old-token"}
-    expected_post_body = {"friendlyName": "test-api-key-name_12345", "expireDays": 1}
-    mock_requests.post.assert_called_with(
-        "https://api.dm.iss-scms.com/api/test-token",
-        json=expected_post_body,
-        headers=expected_headers,
-    )
-
-    expected_delete_body = {"friendlyName": "test-api-key-name_01234"}
-    mock_requests.delete.assert_called_with(
-        "https://api.dm.iss-scms.com/api/test-token",
-        json=expected_delete_body,
-        headers=expected_headers,
-    )
-
-    # Assert final value
-    assert actual_value == expected_value
+from unittest.mock import patch, MagicMock
+import os
+import json
+
+from addons.images.iss_health_check import iss_token
+
+# --------------------- Storage Type tests ---------------------
+
+@patch.dict(
+    os.environ,
+    {
+        "STORAGE_TYPE": "gcp",
+    },
+)
+def test_get_storage_type_gcp():
+    actual_value = iss_token.get_storage_type()
+    assert actual_value == "gcp"
+
+
+@patch.dict(
+    os.environ,
+    {
+        "STORAGE_TYPE": "postgres",
+    },
+)
+def test_get_storage_type_postgres():
+    actual_value = iss_token.get_storage_type()
+    assert actual_value == "postgres"
+
+
+@patch.dict(
+    os.environ,
+    {
+        "STORAGE_TYPE": "GCP",
+    },
+)
+def test_get_storage_type_gcp_case_insensitive():
+    actual_value = iss_token.get_storage_type()
+    assert actual_value == "gcp"
+
+
+@patch.dict(
+    os.environ,
+    {
+        "STORAGE_TYPE": "POSTGRES",
+    },
+)
+def test_get_storage_type_postgres_case_insensitive():
+    actual_value = iss_token.get_storage_type()
+    assert actual_value == "postgres"
+
+
+@patch.dict(
+    os.environ,
+    {
+        "STORAGE_TYPE": "test",
+    },
+)
+def test_get_storage_type_invalid():
+    try:
+        iss_token.get_storage_type()
+        assert False

```

Comment:
What about using the `with pytest.raises(SystemExit) as err:` instead of asserting inside the raised exception? Feels a bit more proper.

Reply:
https://github.com/Trihydro/jpo-cvmanager/pull/7 has been merged, so this should be resolved now.